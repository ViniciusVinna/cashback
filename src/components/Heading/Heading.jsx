import React from 'react';
import PropTypes from 'prop-types';
import { HeadingStyled } from './Heading.styled';

const {
  H1,
  H2,
  H3,
  H4,
  H5,
  H6,
} = HeadingStyled;

const Heading = ({ children, level, ...rest }) => {
  if (level === 'h2') {
    return (<H2 {...rest}>{children}</H2>);
  }
  else if (level === 'h2') {
    return (<H3 {...rest}>{children}</H3>);
  }
  else if (level === 'h3') {
    return (<H4 {...rest}>{children}</H4>);
  }
  else if (level === 'h4') {
    return (<H5 {...rest}>{children}</H5>);
  }
  else if (level === 'h5') {
    return (<H6 {...rest}>{children}</H6>);
  }

  return (<H1 {...rest}>{children}</H1>);
};

Heading.propTypes = {
  children: PropTypes.node.isRequired,
  level: PropTypes.string,
};

Heading.defaultProps = {
  level: 'h1',
};

export default Heading;
