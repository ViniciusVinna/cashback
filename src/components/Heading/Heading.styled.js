import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const HeadingStyled = {
  H1: styled('h1')(({ theme }) => css`
    font-size: ${theme.textSize.xxxxl};
    font-weight: ${theme.font.headingWeight};
    letter-spacing: ${theme.font.headingSpacing};
    line-height: ${theme.lineHeight.s};
    margin: 0 0 ${theme.space.l} 0;
    font-family: ${theme.font.heading};
  `),
  H2: styled('h2')(({ theme }) => css`
    font-size: ${theme.textSize.xxxl};
    font-weight: ${theme.font.headingWeight};
    letter-spacing: ${theme.font.headingSpacing};
    margin: 0 0 ${theme.space.l} 0;
    font-family: ${theme.font.heading};
  `),
  H3: styled('h3')(({ theme }) => css`
    font-size: ${theme.textSize.xxl};
    font-weight: ${theme.font.headingWeight};
    letter-spacing: ${theme.font.headingSpacing};
    line-height: ${theme.lineHeight.s};
    margin: 0 0 ${theme.space.l} 0;
    font-family: ${theme.font.heading};
  `),
  H4: styled('h4')(({ theme }) => css`
    font-size: ${theme.textSize.xl};
    font-weight: ${theme.font.headingWeight};
    letter-spacing: ${theme.font.headingSpacing};
    line-height: ${theme.lineHeight.s};
    margin: 0 0 ${theme.space.l} 0;
    font-family: ${theme.font.heading};
  `),
  H5: styled('h5')(({ theme }) => css`
    font-size: ${theme.textSize.l};
    font-weight: ${theme.font.headingWeight};
    letter-spacing: ${theme.font.headingSpacing};
    line-height: ${theme.lineHeight.s};
    margin: 0 0 ${theme.space.l} 0;
    font-family: ${theme.font.heading};
  `),
  H6: styled('h6')(({ theme }) => css`
    font-size: ${theme.textSize.m};
    font-weight: ${theme.font.headingWeight};
    letter-spacing: ${theme.font.headingSpacing};
    line-height: ${theme.lineHeight.s};
    margin: 0 0 ${theme.space.l} 0;
    font-family: ${theme.font.heading};
  `),
};
