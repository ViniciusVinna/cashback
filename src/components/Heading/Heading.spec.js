import React from 'react';
import Heading from './Heading';

describe('Heading', () => {
  it('it should render the h1 html element by default', () => {
    const { container } = render(<Heading>Heading 1 Title</Heading>);

    expect(container).toMatchSnapshot();
  });

  it('it should render the h2 html element by default', () => {
    const { container } = render(
      <Heading level="h2">Heading 2 Title</Heading>
    );

    expect(container).toMatchSnapshot();
  });

  it('it should render the h3 html element by default', () => {
    const { container } = render(
      <Heading level="h3">Heading 3 Title</Heading>
    );

    expect(container).toMatchSnapshot();
  });

  it('it should render the h4 html element by default', () => {
    const { container } = render(
      <Heading level="h4">Heading 4 Title</Heading>
    );

    expect(container).toMatchSnapshot();
  });

  it('it should render the h5 html element by default', () => {
    const { container } = render(
      <Heading level="h5">Heading 5 Title</Heading>
    );

    expect(container).toMatchSnapshot();
  });

  it('it should render the h6 html element by default', () => {
    const { container } = render(
      <Heading level="h6">Heading 6 Title</Heading>
    );

    expect(container).toMatchSnapshot();
  });
});
