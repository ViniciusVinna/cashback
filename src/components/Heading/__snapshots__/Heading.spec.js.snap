// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading it should render the h1 html element by default 1`] = `
<div>
  <h1
    class="css-190evkz"
  >
    Heading 1 Title
  </h1>
</div>
`;

exports[`Heading it should render the h2 html element by default 1`] = `
<div>
  <h2
    class="css-icz1uc"
  >
    Heading 2 Title
  </h2>
</div>
`;

exports[`Heading it should render the h3 html element by default 1`] = `
<div>
  <h3
    class="css-1j6gc3q"
  >
    Heading 3 Title
  </h3>
</div>
`;

exports[`Heading it should render the h4 html element by default 1`] = `
<div>
  <h4
    class="css-tzbvbu"
  >
    Heading 4 Title
  </h4>
</div>
`;

exports[`Heading it should render the h5 html element by default 1`] = `
<div>
  <h5
    class="css-1oqlkh3"
  >
    Heading 5 Title
  </h5>
</div>
`;

exports[`Heading it should render the h6 html element by default 1`] = `
<div>
  <h6
    class="css-1b7lpw4"
  >
    Heading 6 Title
  </h6>
</div>
`;
